### ISSUE TITLE
Only default task lists will be created

### ISSUE THREAD

**Issue by amroncz** [RogerSelwyn/O365-HomeAssistant#232]
After I have installed and configured the integration, only the default task lists can be found in my o365_tasks_<user>.yaml:
```
- name: Tasks
  show_completed: false
  task_list_id: <task_list_id>
  track: true

- name: Flagged Emails
  show_completed: false
  task_list_id: <task_list_id>
  track: true
```
My custom shopping list, which I share with other people in the family, was not created.

I tried to add it myself, but that didn't work either.

```
- name: Tasks
  show_completed: false
  task_list_id: <task_list_id>
  track: true

- name: Flagged Emails
  show_completed: false
  task_list_id: <task_list_id>
  track: true

- name: Shopping list
  show_completed: false
  task_list_id: <task_list_id>
  track: true
```

How do I add more task lists?

**Comment by RogerSelwyn** [comment]
Do the other lists show when you query ms graph api?

**Comment by amroncz** [comment]
> Do the other lists show when you query ms graph api?

Thank you very much for your quick response. Unfortunately, I'm not that familiar with the subject yet. What exactly do you mean?
In the Microsoft To-Do app and in the browser version, the task lists are displayed.

Übersetzt mit DeepL https://www.deepl.com/app/?utm_source=android&utm_medium=app&utm_campaign=share-translation

**Comment by RogerSelwyn** [comment]
Can you describe how you have added in the MS ToDo app. I have a feeling that shared todo lists aren’t supported.

**Comment by RogerSelwyn** [comment]
See this issue - https://github.com/RogerSelwyn/O365-HomeAssistant/issues/109

**Comment by amroncz** [comment]
> Can you describe how you have added in the MS ToDo app. I have a feeling that shared todo lists aren’t supported.

> See this issue - #109

Are you really sure we're talking about the same thing?

I simply clicked on the “New list” button at the bottom left of the Microsoft Todo app to create another list next to the existing “Tasks” and “Flagged emails” lists. I then shared this list via an invitation link.

For comparison, I have just created another list exactly as described above and this time NOT shared and this list is also not used by the integration.



I'm not sure right now, but I strongly believe that I had already tested the integration some time ago and the corresponding list was taken over by the integration at that time. At that time, however, HA could not do anything with the lists by default and the lovelace card from fixtSE did not yet exist at that point, which is why I did not pursue the whole thing any further.

**Comment by RogerSelwyn** [comment]
I've just gone through and added a todo list via the iOS Reminder app and via the MS Todo App on Mac. After a reboot of HA, they both appear fine in the HA ToDo panel.

**Comment by RogerSelwyn** [comment]
I assume you have `todo_sensors` --> `track_new` to to true?

**Comment by amroncz** [comment]
> I assume you have `todo_sensors` --> `track_new` to to true?

According to the documentation, the default value for this is true, so I didn't specify it explicitly again.

> I've just gone through and added a todo list via the iOS Reminder app and via the MS Todo App on Mac. After a reboot of HA, they both appear fine in the HA ToDo panel.

I have just commented out all lists in o365_tasks_.yaml and restarted HA completely. Only the “Tasks” and “Flagged Emails” lists were added again.

**Comment by amroncz** [comment]
I have just noticed that I have the same problem with the calendars.
“Calendar” and “Birthdays” were added by the integration. While “Holidays” and “Garbage collection” were not added.

Is any value set by default so that only two calendars / task lists are ever added?

**Comment by amroncz** [comment]
I was able to fix the problem.

I did some googling back and forth and it seems like there are a few people here and there who had the same problem. Some had to open the browser version of Outlook once after creating new calendar / task lists. That didn't work for me.
Others had the problem that not all lists were displayed under the API endpoint https://graph.microsoft.com/v1.0/me/todo/lists/, but when they searched for the specific list, they got it from the Graph API. This was not the case for me, so I played around a bit with the Graph API in Postman and even there I only ever got two calendar / task lists. So I created the Azure App registration again exactly as I had done before (using screenshots of the old Azure App registration), deleted the integration and all dependencies, restarted HA, freshly reinstalled the integration and restarted HA again. After that it finally worked.

Since the new Azure App registration looks the same as the old one, I can't say what it was ultimately due to, but if someone has the same problem in the future, they now know that it can help to start from the beginning again.

The issue can be closed. Many thanks for your efforts.
